clang-tools-extra/test/clang-tidy/checkers/bugprone/tagged-union-member-count.cpp

```diff
@@ -0,0 +1,181 @@
+// RUN: %check_clang_tidy %s bugprone-tagged-union-member-count %t
+
+enum tags3 {
+	tags3_1,
+	tags3_2,
+	tags3_3,
+};
+
+enum tags4 {
+	tags4_1,
+	tags4_2,
+	tags4_3,
+	tags4_4,
+};
+
+enum tags5 {
+	tags5_1,
+	tags5_2,
+	tags5_3,
+	tags5_4,
+	tags5_5,
+};
+
+union union3 {
+	short *shorts;
+	int *ints;
+	float *floats;
+};
+
+union union4 {
+	short *shorts;
+	double *doubles;
+	int *ints;
+	float *floats;
+};
+
+// It is not obvious which enum is the tag for the union.
+struct taggedunion2 { // No warnings expected.
```

Comment:
Please pay attention to the naming. This is an invalid testcase, so it should be called `MaybeTaggedUnion`, or something similar. Similarly for the other `struct`s, the naming should reflect what we attempt to test there, from `taggedunionN` it is not obvious.